File: abundant_etc.swift
// http://rosettacode.org/wiki/Abundant,_deficient_and_perfect_number_classifications

func findDivisors(n:Int) -> [Int] {
  var divisors:[Int] = []
  for var i = 1; i < n; i++ {
    if n % i == 0 {
      divisors.append(i)
    }
  }
  return divisors
}

func classify(n:Int) -> Int {
  if findDivisors(n).reduce(0,+) < n {
    return -1
  } else if findDivisors(n).reduce(0,+) == n {
    return 0
  } else if findDivisors(n).reduce(0,+) > n {
    return 1
  } else {
    return 5
  }
}

var deficients:[Int] = []
var abundants:[Int] = []
var perfects:[Int] = []
for var i = 1; i <= 20000; i++ {
  let kind = classify(i)
  if kind == -1 {
    deficients.append(i)
  } else if kind == 1 {
    abundants.append(i)
  } else if kind == 0 {
    perfects.append(i)
  }
}

println(deficients.count)
println(perfects.count)
println(abundants.count)

File: abundant_etc.c
// find deficient, perfect, and abundant numbers in c

#include <stdio.h>

int countDivisors(int n)
{
    int accumulator = 0;
    for (int i = 1; i < n; i++) {
        if (n % i == 0) {
            accumulator += i;
        }
    }
    return accumulator;
}

int classify(int n)
{
    if (countDivisors(n) < n) {
        return -1;
    } else if (countDivisors(n) == n) {
        return 0;
    } else if (countDivisors(n) > n) {
        return 1;
    } else {
        return 5;
    }
}

int main(int argc, const char * argv[]) {
    int deficients = 0;
    int abundants = 0;
    int perfects = 0;
    for (int i = 1; i <= 20000; i++) {
        if (classify(i) == -1) {
            deficients++;
        } else if (classify(i) == 0) {
            perfects++;
        } else if (classify(i) == 1) {
            abundants++;
        } else {
            continue;
        }
    }
    printf("deficients: %d\n", deficients);
    printf("perfects: %d\n", perfects);
    printf("abundants: %d\n", abundants);
    return 0;
}
